### PR TITLE
fix(eval): improve security of safe-eval

### PR DESCRIPTION
**Pull request by 80avin**
## PR description

* block reading properties 'constructor', '__proto__', '__defineGetter__', '__defineSetter__' if they are not owned by the object.
* allow only expected variables in global scope ( removing constructor, __proto__, etc from global scope )
* Remove previous patches to fix security issues. Ensure no breakage by adding unit tests

## Checklist

- [x] - Added tests
- [x] - Ran `npm test`, ensuring linting passes
- [x] - Adjust README documentation if relevant

### CODE REVIEW

**Comment by 80avin** [comment]
@brettz9 Please review.
Hope this will fix most of the security issues.

File: test/test.safe-eval.js [hunk 1]
```diff
@@ -1,77 +1,98 @@
-import {checkBuiltInVMAndNodeVM} from '../test-helpers/checkVM.js';
+import {checkBuiltInVMAndNodeVM} from "../test-helpers/checkVM.js";
```
Comment (brettz9) [inline test/test.safe-eval.js:1]: Is this passing linting? The linting should normally be enforcing single quotes...

Comment (80avin) [reply test/test.safe-eval.js:1]: Yes, it is passing lint and was done by autoformat only.
I also noticed that some files use `"` ( https://github.com/JSONPath-Plus/JSONPath/blob/39ee1844c610f9fd8925d3095ed3971e3d48025e/test/test.cli.js#L1 ) while some use `'` ( https://github.com/JSONPath-Plus/JSONPath/blob/39ee1844c610f9fd8925d3095ed3971e3d48025e/test/test.eval.js#L1 ), suggesting that linting/formatting tooling is broken or not strictly defined.

I tried checking out to very old commits (from 2022) and using different editors (neovim, vscode), it still formats the same.

Comment (80avin) [reply test/test.safe-eval.js]: I've removed the unnecessary formatting changes in this file. We can handle format/lint issue separately.
Also updated the docs & badges.

**Comment by brettz9** [comment]
@80avin : Please check your email . Thanks!

**Pull request merged**